src/mono/mono/mini/method-to-ir.c

```diff
@@ -12062,6 +12062,25 @@ mono_op_no_side_effects (int opcode)
 	}
 }
 
+gboolean
+mono_ins_no_side_effects (MonoInst *ins)
+{
+	if (mono_op_no_side_effects (ins->opcode))
+		return TRUE;
+	if (ins->opcode == OP_AOTCONST) {
+		MonoJumpInfoType type = (MonoJumpInfoType)ins->inst_p1;
+		// Some AOTCONSTs have side effects
+		switch (type) {
+		case MONO_PATCH_INFO_TYPE_FROM_HANDLE:
```

Comment:
When does assembly loading happen? By the time we need to resolve this patch everything is already loaded?

Reply:
Loading happens when the aot-ed method is loaded/first called.